### PR TITLE
Add ramips mt7628an Keenetic-4g

### PR DESCRIPTION
**Pull request by frol1805**

### CODE REVIEW

**Comment by adrianschmutzler** [comment]
Hi, please have a look at our formal requirements at https://openwrt.org/submitting-patches

File: _config.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+theme: jekyll-theme-architect
```
Comment (adrianschmutzler) [inline _config.yml:1]: Remove this file.

File: target/linux/ramips/mt76x8/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -79,6 +79,10 @@ ramips_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"1:lan" "2:lan" "3:lan" "4:lan" "0:wan" "6@eth0"
 		;;
+	zyxel,keenetic-4g)
```
Comment (adrianschmutzler) [inline target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:82]: Sorting.

File: target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds [hunk 3]
```diff
@@ -135,6 +135,11 @@ zyxel,keenetic-extra-ii)
 	set_wifi_led "$boardname:green:wifi"
 	ucidef_set_led_switch "internet" "internet" "$boardname:green:internet" "switch0" "0x01"
 	;;
+zyxel,keenetic-4g)
+	set_wifi_led "$boardname:green:wifi"
+	ucidef_set_led_switch "internet" "internet" "$boardname:green:internet" "switch0" "0x01"
+	;;
+esac
```
Comment (adrianschmutzler) [inline target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds:142]: One esac will be enough. Sorting.
Wifi LED should be implemented in device tree.

File: target/linux/ramips/image/mt76x8.mk [hunk 4]
```diff
@@ -618,3 +618,14 @@ define Device/zyxel_keenetic-extra-ii
 	check-size | zyimage -d 6162 -v "ZyXEL Keenetic Extra II"
 endef
 TARGET_DEVICES += zyxel_keenetic-extra-ii
+
+define Device/zyxel_keenetic-4g
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:622]: Sorting.

File: target/linux/ramips/image/mt76x8.mk [hunk 5]
```diff
@@ -618,3 +618,14 @@ define Device/zyxel_keenetic-extra-ii
...
+  IMAGE_SIZE := 16064k
+  BLOCKSIZE := 64k
+  DEVICE_TITLE := Keenetic 4G
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-usb-ledtrig-usbport
+  IMAGES += factory.bin
+  IMAGE/factory.bin := $$(sysupgrade_bin) | pad-to $$$$(BLOCKSIZE) | \
+	check-size  | zyimage -d 4624 -v "KN-4G"
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:629]: Only one space after check-size.

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+		poll-interval = <20>;
+
+		reset {
+            label = "reset";
+            gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+        };
+
+
+		wps {
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts:53]: Please take care of formatting/indent:
- DTS should have only tabs for indent. One empty line should be left between nodes.

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+		wifi {
+			label = "keenetic-4g:green:wifi";
+			gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+		};
+		
+    };
+	
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts:44]: Use gpio-keys and drop poll-interval (just look at the other files for examples.)

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		
+		internet {
+			label = "keenetic-4g:green:internet";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			default-state = "keep";
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts:31]: drop default state.

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+/ {
+	compatible = "zyxel,keenetic-4g", "mediatek,mt7628an-soc";
+	model = "Keenetic 4G";
+
+	aliases {
+		led-boot = "/leds/power";
+		led-failsafe = "/leds/power";
+		led-running = "/leds/power";
+		led-upgrade = "/leds/power";
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts:17]: You do not define a power-led, and the definition here would be different anyway.

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "zyxel,keenetic-4g", "mediatek,mt7628an-soc";
+	model = "Keenetic 4G";
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts:11]: missing Zyxel.

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts (hunk 6 above)
Comment (frol1805) [reply target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts:44]: Fix tabs

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts (hunk 8 above)
Comment (frol1805) [reply target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts:24]: Del "perst"

**Comment by adrianschmutzler** [comment]
As indicated by my comment about the formal requirements, please squash all these commits into a single, proper commit with title, description and Signed-off-by as requested by the guidelines, and then force-push the updated commit to this PR.

File: target/linux/ramips/mt76x8/base-files/etc/board.d/02_network (hunk 2 above)
Comment (frol1805) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:78]: Fix

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts (hunk 7 above)
Comment (frol1805) [reply target/linux/ramips/dts/mt7628an_zyxel_keenetic-4g.dts]: fix

**Comment by frol1805** [comment]
Add support ramips mt7628an Zyxel Keenetic-4g

**Review by frol1805 — COMMENTED** [review]
Add ramips mt7628an Zyxel Keenetic-4g

**Review by frol1805 — COMMENTED** [review]
Fix Zyxel Keenetic-4g

**Pull request closed without merging**